packages/apollo-gateway/src/__tests__/integration/cucumber.test.ts

```diff
@@ -0,0 +1,141 @@
+import gql from 'graphql-tag';
+import { execute, ServiceDefinitionModule } from '../execution-utils';
+
+import * as accounts from '../__fixtures__/schemas/accounts';
+import * as books from '../__fixtures__/schemas/books';
+import * as inventory from '../__fixtures__/schemas/inventory';
+import * as product from '../__fixtures__/schemas/product';
+import * as reviews from '../__fixtures__/schemas/reviews';
+
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import { DocumentNode } from 'graphql';
+import { GraphQLExecutionResult } from 'apollo-server-types';
+import { QueryPlan } from '../..';
+
+import { astSerializer, queryPlanSerializer } from '../../snapshotSerializers';
+import prettyFormat from 'pretty-format';
+
+const providesFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/integration/provides.feature',
+);
+const abstractTypesFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/integration/abstract-types.feature',
+);
+
+let serviceFixtures: { [key: string]: ServiceDefinitionModule } = {
+  accounts,
+  books,
+  inventory,
+  product,
+  reviews,
+};
+
+const features = [
+  // providesFeature,
+  abstractTypesFeature,
+];
+
+features.forEach((feature) => {
```

Comment:
i think this is complex enough that we should have tests for it, which kinda raises the question of whether it's something we should be requiring implementers to do at all. is there any way we can simplify the interface?